### PR TITLE
8266318: Switch to macos prefix for macOS bundles

### PR DESCRIPTION
**Pull request by GoeLin**
I backport this for parity with 11.0.16-oracle.

platform.m4: Resolved copyright.
jib-proviles.js: macos-aarch64 not listed. Skipped.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266318](https://bugs.openjdk.java.net/browse/JDK-8266318): Switch to macos prefix for macOS bundles

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Vladimir Kempik](https://openjdk.java.net/census#vkempik) (@VladimirKempik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/958/head:pull/958` \
`$ git checkout pull/958`

Update a local copy of the PR: \
`$ git checkout pull/958` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/958/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 958`

View PR using the GUI difftool: \
`$ git pr show -t 958`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/958.diff">https://git.openjdk.java.net/jdk11u-dev/pull/958.diff</a>

</details>

### CODE REVIEW

File: make/conf/jib-profiles.js [hunk 1]
```diff
@@ -611,7 +607,7 @@ var getJibProfilesProfiles = function (input, common, data) {
             platform: "linux-x86",
         },
         "macosx-x64": {
-            platform: "osx-x64",
+            platform: "jacos-x64",
```
Comment (TheRealMDoerr) [inline make/conf/jib-profiles.js:610]: Typo!

**Review by TheRealMDoerr — CHANGES_REQUESTED** [review]
Please fix typo.

**Comment by GoeLin** [comment]
Thanks for spotting this! Fixed.

**Review by TheRealMDoerr — APPROVED** [review]
LGTM.

**Review by VladimirKempik — APPROVED** [review]

**Comment by GoeLin** [comment]
Pre-submit failures
gc/TestNumWorkerOutput.java#id1 known CMS issue on x86_32, unrelated.
Linux x64: build dependencies, unrelated

**Pull request closed without merging**